boards/arm/frdm_kl46z/doc/index.rst

```diff
@@ -0,0 +1,225 @@
+
+Programming and Debugging
+*************************
+
+Build and flash applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Configuring a Debug Probe
+=========================
+
+A debug probe is used for both flashing and debugging the board. This board is
+configured by default to use the :ref:`opensda-daplink-onboard-debug-probe`.
+
+Early versions of this board have an outdated version of the OpenSDA bootloader
+and require an update. Please see the `DAPLink Bootloader Update`_ page for
+instructions to update from the CMSIS-DAP bootloader to the DAPLink bootloader.
+
+Option 1: :ref:`opensda-daplink-onboard-debug-probe` (Recommended)
+------------------------------------------------------------------
+
+Install the :ref:`pyocd-debug-host-tools` and make sure they are in your search
+path.
+
+Follow the instructions in :ref:`opensda-daplink-onboard-debug-probe` to program
+the `OpenSDA DAPLink FRDM-KL46Z Firmware`_.
+
+Option 2: :ref:`opensda-jlink-onboard-debug-probe`
+--------------------------------------------------
+
+Install the :ref:`jlink-debug-host-tools` and make sure they are in your search
+path.
+
+Follow the instructions in :ref:`opensda-jlink-onboard-debug-probe` to program
+the `OpenSDA J-Link FRDM-KL46Z Firmware`_.
+
+Add the argument ``-DOPENSDA_FW=jlink`` when you invoke ``west build`` to
+override the default runner from pyOCD to J-Link:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_kl46z
+   :gen-args: -DOPENSDA_FW=jlink
+   :goals: build
+
+Configuring a Console
+=====================
+
+Regardless of your choice in debug probe, we will use the OpenSDA
+microcontroller as a usb-to-serial adapter for the serial console.
+
+Connect a USB cable from your PC to J7.
+
+Use the following settings with your serial terminal of choice (minicom, putty,
+etc.):
+
+- Speed: 115200
+- Data: 8 bits
+- Parity: None
+- Stop bits: 1
+
+Flashing
+========
+
+Here is an example for the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_kl46z
+   :goals: flash
+
+Open a serial terminal, reset the board (press the SW1 button), and you should
+see the following message in the terminal:
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS build v2.1.0-rc1-142-g0c804b3fc0ab *****
+   Hello World! frdm_kl46z
+
+Debugging
+=========
+
+Here is an example for the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_kl46z
+   :goals: debug
+
+Open a serial terminal, step through the application in your debugger, and you
+should see the following message in the terminal:
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS build zephyr-v2.0.0-1754-g18cbd0dc81e6 *****
+    Hello World! frdm_kl46z
+
+.. _FRDM-KL46Z Website:
```

Comment:
Need to add some more targets for links here (as noted in review comments above)